src/field_theory/perfect_closure.lean

```diff
@@ -153,12 +153,12 @@ instance : comm_monoid (perfect_closure K p) :=
 { mul_assoc := λ e f g, quot.induction_on e $ λ ⟨m, x⟩, quot.induction_on f $ λ ⟨n, y⟩,
     quot.induction_on g $ λ ⟨s, z⟩, congr_arg (quot.mk _) $
     by simp only [add_assoc, mul_assoc, nat.iterate₂ (frobenius_mul _),
-      ← nat.iterate_add, add_comm, add_left_comm],
+      ← nat.iterate_add_apply, add_comm, add_left_comm],
   one := mk K p (0, 1),
   one_mul := λ e, quot.induction_on e (λ ⟨n, x⟩, congr_arg (quot.mk _) $
-    by simp only [nat.iterate₀ (frobenius_one _), nat.iterate_zero, one_mul, zero_add]),
+    by simp only [nat.iterate₀ (frobenius_one _), nat.iterate_zero, id, one_mul, zero_add]),
```

Comment:
Why does this require `id` as a simp lemma now?  There's also the addition of `(∘)` somewhere else.

Reply:
I modified `iterate_zero` to rewrite `f^[0] = id` instead of `f^[0] x = x`. Added `iterate_zero_apply` as suggested by @sgouezel .